### PR TITLE
Feature/external image storage

### PR DESCRIPTION
**Pull request by nilroy**
### Description

This implements a feature to enable external_image_storage in AWS S3. I don't have access to othe cloud platforms so I could implement only this part.

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/grafana/blob/master/TESTING.md -- This file does not exist
- [x] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a <a href="https://github.com/sous-chefs/grafana/blob/master/CHANGELOG.md">CHANGELOG</a> entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: resources/config_external_image_storage.rb
Comment (xorima) [inline resources/config_external_image_storage.rb]: Not familiar here, why would these not be required?

File: resources/config_external_image_storage.rb
Comment (xorima) [inline resources/config_external_image_storage.rb]: If we are supporting multiple providers here then we need to look into naming of variables to work out which are required for which provider, it may be better to create as a specific s3 resource?

File: resources/config_external_image_storage.rb
Comment (xorima) [inline resources/config_external_image_storage.rb]: Please don't require and default, I would remove the default in this case

File: CHANGELOG.md
Comment (xorima) [inline CHANGELOG.md]: Set to unreleased, as version number has not been decided :)

**Review by xorima — CHANGES_REQUESTED** [review]

**Comment by xorima** [comment]
Thanks, looks good just a few comments

File: resources/config_external_image_storage.rb
Comment (nilroy) [reply resources/config_external_image_storage.rb]: @Xorima the path is optional. I modifed the code so that atleast we need to set either bucket/bucket_url. As far as it goes with the access_key and secret_key they are optional. We can rely on the IAM role of the ec2 instance that is running on Grafana. That's why I set those as optional

File: resources/config_external_image_storage.rb
Comment (nilroy) [reply resources/config_external_image_storage.rb]: @Xorima you are correct. Now I have renamed the resource to be S3 specific.

File: resources/config_external_image_storage.rb
Comment (nilroy) [reply resources/config_external_image_storage.rb]: @Xorima fixed these issues.

File: CHANGELOG.md
Comment (nilroy) [reply CHANGELOG.md]: @Xorima I have set it to unreleased :)

File: documentation/grafana_config_external_image_storage_s3.md
Comment (xorima) [inline documentation/grafana_config_external_image_storage_s3.md]: you have a couple of ` here :)

**Review by xorima — DISMISSED** [review]

**Comment by xorima** [comment]
Thanks for this, one tiny change on the docs and we can get this merged 👍

File: documentation/grafana_config_external_image_storage_s3.md
Comment (nilroy) [reply documentation/grafana_config_external_image_storage_s3.md]: @Xorima sorry for that. Fixed it now

**Review by xorima — APPROVED** [review]

**Comment by nilroy** [comment]
@Xorima thanks for merging this. Do you know when I can expect this changes to be in chef supermarket as a new version

**Comment by xorima** [comment]
Waiting on an approval for the release (See #240 ) and then I can release some point this weekend, should be in supermarket by Monday morning, did you need it sooner?

**Comment by nilroy** [comment]
> Waiting on an approval for the release (See #240 ) and then I can release some point this weekend, should be in supermarket by Monday morning, did you need it sooner?

Thanks a lot. Monday works well :) 👍

**Pull request merged**
